### ISSUE TITLE
docs: align 'Ibis for pandas Users' with recommended way of working

### ISSUE THREAD

**Issue by ianmcook** [ibis-project/ibis#6007]
### Describe the solution you'd like

The "Ibis for pandas Users" guide directs readers to use Ibis's pandas backend. Is this aligned with the way that the Ibis developers recommend using Ibis? Should it instead (or additionally) emphasize to readers that they can take a pandas DataFrame, use it to create an Ibis table backed by the default DuckDB backend, and then run Ibis on that?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ianmcook** [comment]
There is some risk in showing users how to create a DuckDB-backed Ibis table directly from a pandas DataFrame, because this might encourage less efficient ways of working, like: Parquet file -> pandas DataFrame -> Ibis table backed by DuckDB, when it would be more efficient to go directly Parquet file -> Ibis table backed by DuckDB. So the guide should explicitly discourage that. But there are plenty of other cases in which the user already has pandas DataFrame in memory and wants to use Ibis to work with it using the most efficient local backend.

**Comment by gforsyth** [comment]
> Is this aligned with the way that the Ibis developers recommend using Ibis?

No, it is not.

I think we should demonstrate that if you already have something in pandas, you can start from there and continue with Ibis.  But that should be balanced against using the DuckDB backend directly.

**Comment by ianmcook** [comment]
Digging into this deeper: there are a couple of options available for a user who has a pandas DataFrame `df` in memory and wants to use Ibis to query it with the embedded DuckDB backend:

Option 1:
```py
t = ibis.memtable(df)
```

Option 2:
```py
conn = ibis.duckdb.connect(':memory:')
conn.create_table('name', df)
t = conn.table('name')
```

In Option 1, I believe no data is copied when `t` is created, and DuckDB scans the pandas DataFrame when the query is executed. In Option 2, I believe the data is copied from the pandas DataFrame into a transient DuckDB in-memory table when `t` is created. Please correct me if any of that is wrong.

**Comment by ianmcook** [comment]
Also: I believe this is shorthand for Option 2:
```py
t = ibis.memtable(df).cache()
```

**Comment by jcrist** [comment]
Thanks for opening this Ian. In general I think this doc should cover two things:

- How to load data into ibis for traditional pandas tasks. This would cover loading from local files (csv & parquet), and from existing pandas dataframes
- How ibis expressions and data models compare with pandas's operations. A side-by-side comparison for common tasks. The backend used here doesn't really matter since we're just covering the expression level, but we should probably use duckdb since it's the best local backend.

----

As for your specific operation questions:

- `ibis.memtable` will use a view in duckdb, no data is copied.
- `create_table` will create a new table, which will copy the data into duckdb's internal format
- If you have access to a `con` already, `table = con.register(df)` will also create a view (no data copied), but will get you an actual `Table` instead of a `memtable`. This should be functionally equivalent to `ibis.memtable`, either option is fine.
- `table.cache()` will create a temp table, which will copy the data but won't persist the table after the connection is closed.

**Comment by jcrist** [comment]
Fixed by #6048. Closing.
